test/explain/algorithm/test_gnn_explainer.py

```diff
@@ -0,0 +1,390 @@
+    model,
+    return_type,
+    explanation_type,
+):
+    explainer_config = ExplainerConfig(
+        explanation_type=explanation_type,
+        node_mask_type=node_mask_type,
+        edge_mask_type=edge_mask_type,
+    )
+    model_config = ModelConfig(
+        task_level='node',
+        return_type=return_type,
+        mode="classification",
+    )
+
+    model = model()
+
+    out = model(x, edge_index).argmax(dim=-1)
+
+    if isinstance(model, GCN_classification_multi_output):
+        target_index = 0
+    else:
+        target_index = None
+
+    explainer = Explainer(model=model, algorithm=GNNExplainer(epochs=2),
+                          explainer_config=explainer_config,
+                          model_config=model_config)
+
+    # try to explain prediction for node 0
+    explanation = explainer(x=x, edge_index=edge_index, target=out, index=2,
+                            target_index=target_index)
+
+    check_explanation(edge_mask_type, node_mask_type, x, edge_index,
+                      explanation)
+
+
+@pytest.mark.parametrize('edge_mask_type', edge_mask_types)
+@pytest.mark.parametrize('node_mask_type', node_mask_types)
+@pytest.mark.parametrize("explanation_type", ["model", "phenomenon"])
+@pytest.mark.parametrize(
+    'model', [GCN_regression_single_output, GCN_regression_multi_output])
+@pytest.mark.parametrize('return_type', return_types_regression)
+def test_gnn_explainer_with_meta_explainer_regression_node(
+    edge_mask_type,
+    node_mask_type,
+    model,
+    return_type,
+    explanation_type,
+):
+    explainer_config = ExplainerConfig(
+        explanation_type=explanation_type,
+        node_mask_type=node_mask_type,
+        edge_mask_type=edge_mask_type,
+    )
+    model_config = ModelConfig(
+        task_level='node',
+        return_type=return_type,
+        mode="regression",
+    )
+
+    model = model()
+    with torch.no_grad():
+        out = model(x, edge_index)
+
+    if isinstance(model, GCN_regression_multi_output):
+        target_index = 0
+    else:
+        target_index = None
+
+    explainer = Explainer(model=model, algorithm=GNNExplainer(epochs=2),
+                          explainer_config=explainer_config,
+                          model_config=model_config)
+
+    # try to explain prediction for node 0
+    explanation = explainer(x=x, edge_index=edge_index, target=out, index=2,
+                            target_index=target_index)
+
+    check_explanation(edge_mask_type, node_mask_type, x, edge_index,
+                      explanation)
+
+
+@pytest.mark.parametrize('edge_mask_type', edge_mask_types)
+@pytest.mark.parametrize('node_mask_type', node_mask_types)
+@pytest.mark.parametrize("explanation_type", ["model", "phenomenon"])
+@pytest.mark.parametrize('model', [
+    GNN_regression_single_output,
+    GNN_regression_multi_output,
+])
+@pytest.mark.parametrize('return_type', return_types_regression)
+def test_gnn_explainer_with_meta_explainer_regression_graph(
```

Comment:
Put test_classification_graph function before test_regression_graph to be consistent with node test order.